src/controllers/trainee/controller.ts

```diff
@@ -0,0 +1,81 @@
+class TraineeController {
+    static instance: TraineeController;
+
+    static getInstance() {
+         if ( TraineeController.instance) {
+             return TraineeController.instance;
+         }
+         TraineeController.instance = new TraineeController();
+         return TraineeController.instance;
+    }
+    get (req : REQUEST, res : RESOURCE, next : NEXTFUNCTION ) {
```

Comment:
You have not exported the REQUEST, RESPONSE, NEXTFUNCTION ts compiler will trrow error for this.
You need to import  {Request, Response, NextFunction } from 'express'; 
Fix this in all function where you use this (REQUEST, RESPONSE, NEXTFUNCTION) .